test/unit/resources/aws_config_recorder_test.rb

```diff
@@ -0,0 +1,153 @@
+    assert_equal('default', AwsConfigurationRecorder.new(recorder_name: 'default').recorder_name)
+  end
+
+  def test_property_role_arn
+    assert_equal('arn:aws:iam::721741954427:role/default', AwsConfigurationRecorder.new(recorder_name: 'default').role_arn)
+    assert_nil(AwsConfigurationRecorder.new(recorder_name: 'NonExistentRecorder').role_arn)
+  end
+  
+  def test_property_resource_types
+    assert_equal(['AWS::EC2::CustomerGateway', 'AWS::EC2::EIP'], AwsConfigurationRecorder.new(recorder_name: 'Recorder_2').resource_types)
+    assert_nil(AwsConfigurationRecorder.new(recorder_name: 'NonExistentRecorder').resource_types)
+  end
+end
+
+#=============================================================================#
+#                               Test Matchers
+#=============================================================================#
+class AwsConfigurationRecorderPropertiesTest < Minitest::Test
+  def test_matcher_all_supported
+    assert AwsConfigurationRecorder.new(recorder_name: 'default').all_supported?
+    refute AwsConfigurationRecorder.new(recorder_name: 'Recorder_1').all_supported?
+  end
+  
+  def test_matcher_has_include_global_resource_types
+    assert AwsConfigurationRecorder.new(recorder_name: 'default').has_include_global_resource_types?
+    refute AwsConfigurationRecorder.new(recorder_name: 'Recorder_1').has_include_global_resource_types?
+  end
+  
+  def test_matcher_recording
+    assert AwsConfigurationRecorder.new(recorder_name: 'default').recording?
+    refute AwsConfigurationRecorder.new(recorder_name: 'Recorder_1').recording?
+  end
+end
+
+
+#=============================================================================#
+#                               Test Fixtures
+#=============================================================================#
+
+module AwsMCRSB
+  class Basic < AwsBackendBase
+    def describe_configuration_recorders(query)
+      recorders = {
+        'default' => OpenStruct.new({
+          :configuration_recorders => [
+            name: "default",
+            role_arn: "arn:aws:iam::721741954427:role/default",
+            :recording_group => OpenStruct.new({
+              all_supported: true,
+              include_global_resource_types: true,
+              resource_types: []
+            }),
+          ]
+        }),
+        'Recorder_2' => OpenStruct.new({
+          :configuration_recorders => [
+            name: "Recorder_2",
+            role_arn: "arn:aws:iam::721741954427:role/Recorder_1",
+            :recording_group => OpenStruct.new({
+              all_supported: false,
+              include_global_resource_types: false,
+              resource_types: ['AWS::EC2::CustomerGateway', 'AWS::EC2::EIP']
+            }),
+          ]
+        }),
+        'empty' => {}
+      }
+      return recorders[query[:configuration_recorder_names][0]] unless recorders[query[:configuration_recorder_names][0]].nil?
+      recorders['empty']
```

Comment:
You should raise `Aws::ConfigService::Errors::NoSuchConfigurationRecorderException` here